src/main/features/core/tray.js

```diff
@@ -40,9 +63,35 @@ const setContextMenu = () => {
 setContextMenu();
 
 appIcon.setToolTip('Google Play Music');
-appIcon.on('double-click', () => {
-  mainWindow.setSkipTaskbar(false);
-  mainWindow.show();
+
+
+// Iconified     -> show, to foreground
+// In background -> to foreground
+// In foreground -> iconify
+appIcon.on('click', () => {
+  if (process.platform === 'darwin') {
+    // Special code for OS-X - Not tested!
+    if (!mainWindow.isVisible()) {
+      mainWindow.setSkipTaskbar(false);
+      mainWindow.show();
+    } else {
+      if (PlaybackAPI.isPlaying()) {
+        Emitter.sendToGooglePlayMusic('playback:playPause');
+      }
+      mainWindow.hide();
+    }
+  } else {
+    // Windows, Linux
+    if (mainWindow.isMinimized() || !mainWindow.isFocused()) {
+      // Window in tray, or not focused
+      mainWindow.setSkipTaskbar(false);
+      mainWindow.show();
+      mainWindow.focus();
+    } else {
+      mainWindow.minimize();
```

Comment:
We shouldn't be hiding the player when clicking the tray icon without checking `Settings.get('minToTray`, true)`